src/transformers/hf_argparser.py

```diff
@@ -12,27 +12,32 @@ class HfArgumentParser(ArgumentParser):
     """
     This subclass of `argparse.ArgumentParser` uses type hints on dataclasses
     to generate arguments.
+
+    The class is designed to play well with the native argparse. In particular,
+    you can add more (non-dataclass backed) arguments to the parser after initialization
+    and you'll get the output back after parsing as an additional namespace.
     """
 
-    def __init__(self, obj: Union[None, DataClass, DataClassType] = None):
-        """
-        Args:
-            obj:
-                (Optional) Can be either a dataclass instance, or a dataclass type.
-                Both will work the exact same way.
-        """
-        super().__init__()
-        if obj is not None:
-            self.add_dataclass_arguments(obj)
+    dataclass_types: Iterable[DataClassType]
 
-    def add_dataclass_arguments(self, obj: Union[DataClass, DataClassType]):
+    def __init__(self, dataclass_types: Union[DataClassType, Iterable[DataClassType]], **kwargs):
         """
         Args:
-            obj:
-                Can be either a dataclass instance, or a dataclass type.
-                Both will work the exact same way.
+            dataclass_types:
+                Dataclass type, or list of dataclass types for which we will "fill" instances
+                with the parsed args.
+            kwargs:
+                (Optional) Passed to `argparse.ArgumentParser()` in the regular way.
         """
-        for field in dataclasses.fields(obj):
+        super().__init__(**kwargs)
+        if dataclasses.is_dataclass(dataclass_types):
+            dataclass_types = [dataclass_types]
+        self.dataclass_types = dataclass_types
+        for dtype in self.dataclass_types:
+            self._add_dataclass_arguments(dtype)
+
+    def _add_dataclass_arguments(self, dtype: DataClassType):
+        for field in dataclasses.fields(dtype):
```

Comment:
Ok, final set of improvements before merging! 
- Everything is done at init time.
- You can add_argument using the standard method (not dataclasses backed) on the parser and you'll get the resulting object at the end of the tuple when calling `parse_args_into_dataclasses()`